starknet-rpc-test/get_class_at.rs

```diff
@@ -0,0 +1,116 @@
+use starknet_core::types::contract::SierraClass;
+use starknet_core::types::{BlockId, ContractClass, FlattenedSierraClass, StarknetError};
+use starknet_ff::FieldElement;
+use starknet_providers::ProviderError::StarknetError as StarknetProviderError;
+use starknet_providers::{MaybeUnknownErrorCode, Provider, StarknetErrorWithMessage};
+use starknet_rpc_test::constants::{CAIRO_1_ACCOUNT_CONTRACT, TEST_CONTRACT_ADDRESS};
+use starknet_rpc_test::{ExecutionStrategy, MadaraClient};
+
+#[tokio::test]
+async fn fail_non_existing_block() -> Result<(), anyhow::Error> {
+    let madara = MadaraClient::new(ExecutionStrategy::Native).await;
+    let rpc = madara.get_starknet_client();
+    let test_contract_address = FieldElement::from_hex_be(TEST_CONTRACT_ADDRESS).expect("Invalid Contract Address");
+
+    assert_matches!(
+        rpc
+        .get_class_at(
+            BlockId::Number(100),
+            test_contract_address,
+        )
+        .await,
+        Err(StarknetProviderError(StarknetErrorWithMessage { code: MaybeUnknownErrorCode::Known(code), .. })) if code == StarknetError::BlockNotFound
+    );
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn fail_non_existing_contract() -> Result<(), anyhow::Error> {
+    let madara = MadaraClient::new(ExecutionStrategy::Native).await;
+    let rpc = madara.get_starknet_client();
+    let unknown_contract_address = FieldElement::from_hex_be("0x4269DEADBEEF").expect("Invalid Contract Address");
+
+    assert_matches!(
+        rpc
+        .get_class_at(
+            BlockId::Number(0),
+            unknown_contract_address,
+        )
+        .await,
+        Err(StarknetProviderError(StarknetErrorWithMessage { code: MaybeUnknownErrorCode::Known(code), .. })) if code == StarknetError::ContractNotFound
+    );
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn work_ok_retrieving_class_for_contract_version_0() -> Result<(), anyhow::Error> {
+    let madara = MadaraClient::new(ExecutionStrategy::Native).await;
+    let rpc = madara.get_starknet_client();
+    let test_contract_address = FieldElement::from_hex_be(TEST_CONTRACT_ADDRESS).expect("Invalid Contract Address");
+
+    let test_contract_class_bytes = include_bytes!("../cairo-contracts/build/test.json");
+    let test_contract_class: LegacyContractClass = serde_json::from_slice(test_contract_class_bytes).unwrap();
+
+    assert_matches!(
+        rpc
+        .get_class_at(
+            BlockId::Number(0),
+            test_contract_address
+        ).await?,
+        ContractClass::Legacy(c) => {
+            // decompress program
+            let mut d = GzDecoder::new(&c.program[..]);
+            let mut data = String::new();
+            d.read_to_string(&mut data).unwrap();
+            let program: LegacyProgram = serde_json::from_str(data.as_str())?;
+            assert_eq!(
+                program.data,
+                test_contract_class.program.data,
+            );
+        }
+    );
+
+    Ok(())
+}
+
+#[ignore]
```

Comment:
Add a "TODO: remove when xyz" or open an issue explaining what is buggy and remaining people to un-ignore this tests once it's fixed

Reply:
Issue #1056 added